src/Microsoft.Extensions.Caching.Abstractions/ICacheEntry.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    /// <summary>
+    /// Represents an entry in the <see cref="IMemoryCache"/> implementation.
+    /// </summary>
+    public interface ICacheEntry : IDisposable
+    {
+        /// <summary>
+        /// The key of the <see cref="ICacheEntry"/>.
+        /// </summary>
+        object Key { get; }
+
+        /// <summary>
+        /// Sets the value of the cache entry.
+        /// </summary>
+        /// <param name="value"></param>
+        ICacheEntry SetValue(object value);
+
+        /// <summary>
+        /// Sets the priority for keeping the cache entry in the cache during a memory pressure tokened cleanup.
+        /// </summary>
+        /// <param name="priority"></param>
+        ICacheEntry SetPriority(CacheItemPriority priority);
```

Comment:
This SetFoo, return ICacheEntry style seems out of place on the base interface. We usually keep the interfaces simple and add sugar like this as extensions. I'll keep reviewing and see how it works out.